### ISSUE TITLE
Add `learning_rate` hyper-parameter

### ISSUE THREAD

**Issue by ablaom** [FluxML/MLJFlux.jl#317]
Currently, learning rates are buried in the `optimiser` provided to a model. This is awkward from the point of view of *tuning* the learning rate. One would like to specify a learning rate range like this,

**option 1** `MLJ.range(mljflux_model, :learning_rate, lower=1e-6, upper=1e-3, scale=:log10)`

but instead is limited to

**option 2**

```julia
optimisers = [Adam(10^i) for i in -6:-3]
MLJ.range(mljflux_model, :optimiser, values=optimisers)
```

This is limiting for many reasons. Here are two:

- In option 2 I can't randomly search the entire continuous range I'm interested in (sample from the uniform distribution, on the log scale, as `RandomSearch` will do in option 1

- Plots don't get done on a log scale without intervention, in option 1, and the hyper-parameter labels mess up the standard `plot(tuned_machine)` output. 

One solution is to add `learning_rate` as a new hyper-parameter with the understanding that, when specified, the specified `optimiser` is replaced with a copy in which the learning rate (first positional argument in most cases??) is changed to `learning_rate`.

Thoughts anyone?

**Comment by LucasMatSP** [comment]
For now, MLJFlux.jl supports any rule from Optimisers.jl, right? I wonder if the solution presented might restrict that, namely `OptimiserChain` and decays. Also, couldn't `Optimisers.adjust!` ([link](https://fluxml.ai/Optimisers.jl/stable/api/#Optimisers.adjust!)) be used for this under the hood?

**Comment by LucasMatSP** [comment]
Regardless of the solution we choose, we can't forget to document this clearly. Maybe even with some type of 'warning' block

**Comment by ablaom** [comment]
@LucasMatSP Thanks for those suggestion, and thanks for pointing out `adjust!`. Yes, we can't just assume the learning rate is the the first argument of `optimiser`, as this could be an `OptimiserChain`. How about the user supplies either:

1.  An optimiser as present 
2. A callable `f`, mapping the learning rate to an optimiser. We have a new hyperparameter ,`learning_rate`, and the optimiser used is `f(learning_rate)`.

This design can be implemented with ordinary dispatch if all optimisers we'd want to use, including `OptimiserChain`s,  subtype `Optimisers.AbstractRule`, which appears to be the case. I've posted a query at slack \#machine-learning to confirm.

**Comment by ablaom** [comment]
@LucasMatSP What do you think?

**Comment by LucasMatSP** [comment]
The first option is more convenient for the user. Then, under the hood, `adjust!` would be used to update the learning rate?

**Comment by ablaom** [comment]
Ok, the penny dropped. It looks like `adjust!` will work on any supported optimiser rule state, so we don't need the second option at all. This is pretty straightforward then. Thanks.

**Comment by ablaom** [comment]
Working on this PR, I ran into #322 , which in turn as led to https://github.com/FluxML/Optimisers.jl/pull/224. So waiting on those before proceeding further here, sorry.
